pkg/pulsar/utils/message.go

```diff
@@ -23,10 +23,11 @@ import (
 )
 
 type Message struct {
-	MessageID  MessageID
-	Payload    []byte
-	Topic      string
-	Properties map[string]string
+	MessageID    MessageID
+	Payload      []byte
+	Topic        string
+	Properties   map[string]string
+	PartitionKey string
```

Comment:
```suggestion
	PartitionKey *string
```

is it better?

Reply:
I have considered both of the options, but overall I think it would be better to use `string` type since 1) it make no sense to set the partition key to be the empty string for any pulsar message, and 2) it is more common to see the `string` as the property type instead of the  string pointer.
Still I can make the change if you think it might be better to maintain the same type as the one defined in the `SingleMessageMetadata`.

Reply:
Hi @nodece sorry for bothering, but is there anything I can do to make further progress as regarding the PR?